opensrp-opd/src/main/java/org/smartregister/opd/widgets/OpdBarcodeFactory.java

```diff
@@ -0,0 +1,85 @@
+package org.smartregister.opd.widgets;
+
+import android.content.Context;
+import android.content.Intent;
+import android.view.View;
+
+import com.google.android.gms.vision.barcode.Barcode;
+import com.rengwuxian.materialedittext.MaterialEditText;
+import com.vijay.jsonwizard.constants.JsonFormConstants;
+import com.vijay.jsonwizard.fragments.JsonFormFragment;
+import com.vijay.jsonwizard.interfaces.CommonListener;
+import com.vijay.jsonwizard.interfaces.JsonApi;
+import com.vijay.jsonwizard.interfaces.OnActivityResultListener;
+import com.vijay.jsonwizard.widgets.BarcodeFactory;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.smartregister.opd.listener.LookUpTextWatcher;
+import org.smartregister.opd.utils.OpdConstants;
+
+import java.util.List;
+
+import timber.log.Timber;
+
+import static android.app.Activity.RESULT_OK;
+
+public class OpdBarcodeFactory extends BarcodeFactory {
+
+    private JsonFormFragment jsonFormFragment;
+    private boolean forLookUp;
+
+    @Override
+    public List<View> getViewsFromJson(String stepName, Context context, JsonFormFragment formFragment, JSONObject jsonObject, CommonListener listener) throws Exception {
+        List<View> viewList = super.getViewsFromJson(stepName, context, formFragment, jsonObject, listener);
+        this.jsonFormFragment = formFragment;
+        try {
+            this.forLookUp = jsonObject.has(OpdConstants.KEY.LOOK_UP) &&
+                    jsonObject.get(OpdConstants.KEY.LOOK_UP).toString().equalsIgnoreCase(Boolean.TRUE.toString());
+        } catch (JSONException e) {
+            Timber.e(e);
+            e.printStackTrace();
+        }
+        return viewList;
+    }
+
+    @Override
+    public List<View> getViewsFromJson(String stepName, Context context, JsonFormFragment formFragment, JSONObject jsonObject, CommonListener listener, boolean popup) {
```

Comment:
Add correct NonNull & Nullable annotations for easy usage & null handling in code